api/src/services/db-watcher.js

```diff
@@ -0,0 +1,68 @@
+const EventEmitter = require('events');
+const medicEmitter = new EventEmitter();
+const sentinelEmitter = new EventEmitter();
+const usersEmitter = new EventEmitter();
+
+const db = require('../db');
+const logger = require('../logger');
+
+const OPTS = { live: true, since: 'now', return_docs: false };
+
+const watchMedicChanges = () => {
+  db.medic
+    .changes(OPTS)
+    .on('change', (change) => {
+      medicEmitter.emit('change', change);
+    })
+    .on('error', (err) => {
+      logger.error('Error watching medic changes, restarting: %o', err);
+      process.exit(1);
```

Comment:
This is how the old listener was structured. It had no `complete` event and rage quit on errors. 
I doubt `complete` ever fires on `continuous`.

Reply:
I don't recall ever seeing a timeout from this continuous changes request. I think the parameters that are sent to CouchDb will prevent the timeout: 
```
if (opts.continuous && !('heartbeat' in opts)) {
  opts.heartbeat = DEFAULT_HEARTBEAT;
}
```

So if there is indeed a timeout, it means CouchDb is in real trouble.

Reply:
It would be possible in the future when we have replicas for one couch node to go down but the cluster itself still be available. I assume if one node goes down and it happens to be the one serving this request, then the request will drop but API shouldn't crash.

As this was already an issue, feel free to add this to the replicas issue and we can solve it later.